doc/api/v8.md

```diff
@@ -1051,6 +1051,51 @@ added:
 
 Returns true if the Node.js instance is run to build a snapshot.
 
+### Class: `v8.MemoryProfileWatchdog`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+#### `new MemoryProfileWatchdog([options])`
+
+* `interval` {number} How often the watchdog polls.
+* `maxRss` {number} Node.js will take snapshot when rss is
+  greater than or equal to `maxRss`.
+* `maxUsedHeapSize` {number} Node.js will take snapshot when used heap size is
+  greater than or equal to `maxUsedHeapSize`.
```

Comment:
Should we store that actual trigger (RSS/Heap/Both) and maybe even the values to some location?

Reply:
Good idea.